internal/service/router.go

```diff
@@ -22,27 +24,40 @@ func (s *service) router() chi.Router {
 			handlers.CtxLog(s.log),
 		),
 	)
-	r.Route("/integrations/CertIntegrator", func(r chi.Router) {
+	r.Route("/integrations/CertIntegrator-svc", func(r chi.Router) {
 		// configure endpoints here
 	})
 
 	return r
 }
 
 func (s *service) startListener(ctx context.Context, cfg config.Config) {
-	s.log.Info("Starting issuer listener")
-
-	listeners.NewListener(
-		cfg.Infura().Key,
-		cfg.CertificatesIssuer().Address,
-		cfg.CertificatesIssuer().FromBlock,
-	).Run(ctx, cfg)
-
-	s.log.Info("Starting fabric listener")
-
-	listeners.NewListener(
-		cfg.Infura().Key,
-		cfg.CertificatesFabric().Address,
-		cfg.CertificatesFabric().FromBlock,
-	).Run(ctx, cfg)
+	conn, err := pgx.Connect(ctx, cfg.DBConfig().URL)
+	if err != nil {
+		panic(errors.Wrap(err, "failed to connect to db"))
+	}
+
+	s.log.Info("Starting fabrics listener")
+
+	for _, contract := range cfg.CertificatesFabric().List {
+		listeners.NewListener(
```

Comment:
You spawn a new one goroutine for every contract, but you are able to listen for a batch of addresses, like all Fabrics and all issuers in the 2 goroutine. Every go-ethereum/core/types.Log stores information about the contract address.